roachprod: `init` does not work with local cluster on Linux

On a gceworker:

```
$ roachprod create local -n 3
$ roachprod start local --skip-init
$ roachprod init local

12:47:35 cluster_synced.go:2300: local: initializing cluster
panic: invalid nodeIdx

goroutine 1 [running]:
github.com/cockroachdb/cockroach/pkg/roachprod/vm/local.VMDir({0xc0002c7030?, 0xc00081f7b0?}, 0x4e0085?)
	github.com/cockroachdb/cockroach/pkg/roachprod/vm/local/local.go:46 +0x292
github.com/cockroachdb/cockroach/pkg/roachprod/install.(*SyncedCluster).localVMDir(...)
	github.com/cockroachdb/cockroach/pkg/roachprod/install/cluster_synced.go:123
github.com/cockroachdb/cockroach/pkg/roachprod/install.(*SyncedCluster).NodeDir(0xc00065cb40, 0x8?, 0x1)
	github.com/cockroachdb/cockroach/pkg/roachprod/install/cockroach.go:219 +0x6c
github.com/cockroachdb/cockroach/pkg/roachprod/install.(*SyncedCluster).generateInitCmd(0xc00065cb40, 0x0)
	github.com/cockroachdb/cockroach/pkg/roachprod/install/cockroach.go:679 +0xc9
github.com/cockroachdb/cockroach/pkg/roachprod/install.(*SyncedCluster).initializeCluster(0xc0000b2d80?, {0x12289f0, 0xc00003a0a0}, 0x2?)
	github.com/cockroachdb/cockroach/pkg/roachprod/install/cockroach.go:612 +0x71
github.com/cockroachdb/cockroach/pkg/roachprod/install.(*SyncedCluster).Init(0xc00065cb40, {0x12289f0, 0xc00003a0a0}, 0x0?)
	github.com/cockroachdb/cockroach/pkg/roachprod/install/cluster_synced.go:2301 +0xc6
github.com/cockroachdb/cockroach/pkg/roachprod.Init({0x12289f0, 0xc00003a0a0}, 0xc0006bfb18?, {0x7ffe018f5a90, 0x5})
	github.com/cockroachdb/cockroach/pkg/roachprod/roachprod.go:719 +0x7f
main.glob..func12(0x2?, {0xc0003c73b0?, 0xc0006bfb30?, 0x5a37e5?})
	main/pkg/cmd/roachprod/main.go:488 +0x45
main.wrap.func1(0x18a85e0?, {0xc0003c73b0?, 0x1?, 0x1?})
	main/pkg/cmd/roachprod/main.go:68 +0x3c
github.com/spf13/cobra.(*Command).execute(0x18a85e0, {0xc0003c7390, 0x1, 0x1})
	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:860 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0x18a7be0)
	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra/external/com_github_spf13_cobra/command.go:902
main.main()
	main/pkg/cmd/roachprod/main.go:1044 +0x785
```

Jira issue: CRDB-19735

--- a/pkg/roachprod/install/cluster_synced.go
+++ b/pkg/roachprod/install/cluster_synced.go
@@ -2290,12 +2290,11 @@ func (c *SyncedCluster) ParallelE(
 
 // Init initializes the cluster. It does it through node 1 (as per TargetNodes)
 // to maintain parity with auto-init behavior of `roachprod start` (when
-// --skip-init) is not specified. The implementation should be kept in
-// sync with Start().
+// --skip-init) is not specified.
 func (c *SyncedCluster) Init(ctx context.Context, l *logger.Logger) error {
 	// See Start(). We reserve a few special operations for the first node, so we
 	// strive to maintain the same here for interoperability.
-	const firstNodeIdx = 0
+	const firstNodeIdx = 1
 
 	l.Printf("%s: initializing cluster\n", c.Name)
 	initOut, err := c.initializeCluster(ctx, firstNodeIdx)

--- a/pkg/roachprod/install/cockroach.go
+++ b/pkg/roachprod/install/cockroach.go
@@ -184,27 +184,9 @@ func (c *SyncedCluster) Start(ctx context.Context, l *logger.Logger, startOpts S
 
 		shouldInit := !c.useStartSingleNode()
 		if shouldInit {
-			l.Printf("%s: initializing cluster", c.Name)
-			initOut, err := c.initializeCluster(ctx, node)
-			if err != nil {
-				return nil, errors.WithDetail(err, "unable to initialize cluster")
+			if err := c.Init(ctx, l); err != nil {
+				return nil, errors.Wrap(err, "failed to initialize cluster")
 			}
-
-			if initOut != "" {
-				l.Printf(initOut)
-			}
-		}
-
-		// We're sure to set cluster settings after having initialized the
-		// cluster.
-
-		l.Printf("%s: setting cluster settings", c.Name)
-		clusterSettingsOut, err := c.setClusterSettings(ctx, l, node)
-		if err != nil {
-			return nil, errors.Wrap(err, "unable to set cluster settings")
-		}
-		if clusterSettingsOut != "" {
-			l.Printf(clusterSettingsOut)
 		}
 		return nil, nil
 	})

